dockstore-launcher/src/main/java/io/github/collaboratory/LauncherCWL.java

```diff
@@ -254,9 +257,35 @@ private String createUpdatedInputsAndOutputsJson(Map<String, FileInfo> fileMap,
                 newJSON.put(paramName, newRecord);
 
                 // TODO: fill in for all possible types
-            } else if (currentParam instanceof Integer || currentParam instanceof Float || currentParam instanceof Boolean || currentParam instanceof String
-                                                                    || currentParam instanceof List) {
+            } else if (currentParam instanceof Integer || currentParam instanceof Float || currentParam instanceof Boolean || currentParam instanceof String) {
                 newJSON.put(paramName, currentParam);
+            } else if (currentParam instanceof List) {
+
+                for (int i = 0; i<((List)currentParam).size(); i++) {
+
+                    Map<String, Object> param = ((LinkedHashMap) ((List)currentParam).get(i));
```

Comment:
Simplify this and code starting at line 421 by casting directly and saving it instead of repeatedly casting